crates/uv/tests/pip_sync.rs

```diff
@@ -1100,8 +1100,7 @@ fn mismatched_name() -> Result<()> {
     ----- stdout -----
 
     ----- stderr -----
-    error: Failed to read: foo @ file://[TEMP_DIR]/foo-2.0.1-py3-none-any.whl
-      Caused by: The .dist-info directory tomli-2.0.1 does not start with the normalized package name: foo
+    error: Because foo was found, but has an invalid format and you require foo, we can conclude that the requirements are unsatisfiable.
```

Comment:
This is slightly worse... I guess I could make the "reasons" more specific?

Reply:
Maybe the `.dist-info` thing is so broken that we shouldn't backtrack. IDK.

Reply:
Hm it's a little weird to throw the no solution error this way (i.e. compared to how it's formatted elsewhere)

You could push the detailed reasons into some hints?

Reply:
> Hm it's a little weird to throw the no solution error this way (i.e. compared to how it's formatted elsewhere)

What do you mean by this?

Reply:
```
error: Because foo was found, ...
```

vs

```
× No solution found when resolving dependencies:
      ╰─▶ Because you require
```

I don't mind the lack of the arrow and such but the "No solution found" part seems relevant.

Reply:
Ah yeah. I'll add separately.